apps/SceneClustering/SceneClustering.cpp

```diff
@@ -0,0 +1,315 @@
+
+	uint32_t curr_ID = 0;
+	FOREACH(IdxC, scene.images)
+	{
+		const auto & curr_image = scene.images[IdxC];
+		if(curr_image.IsValid())
+		{
+			view_fwd_reindexing[IdxC] = curr_ID;
+			view_bkwd_reindexing[curr_ID] = IdxC; 
+			nomoko::View v;
+			v.rot = Eigen::Matrix<double,3,3,1>(curr_image.camera.R).cast<float>();
+			v.trans = Eigen::Vector3d(curr_image.camera.C).cast<float>();
+			domset_views.push_back(v);
+			++curr_ID;
+		}
+	}
+
+	FOREACH(IdxP, scene.pointcloud.points)
+	{
+		nomoko::Point p;
+		p.pos = Eigen::Vector3f(scene.pointcloud.points[IdxP]);
+		FOREACH(IdxV, scene.pointcloud.pointViews[IdxP])
+		{
+			auto idx = view_fwd_reindexing[scene.pointcloud.pointViews[IdxP][IdxV]];
+			p.viewList.push_back(idx);
+		}
+		domset_points.push_back(p);
+	}
+
+	nomoko::Domset domset_instance(domset_points, domset_views, domset_cameras, OPT::fVoxelSize);
+
+	domset_instance.clusterViews(OPT::nMinClusterSize, OPT::nMaxClusterSize, OPT::nClusterOverlap);
+
+	#if TD_VERBOSE != TD_VERBOSE_OFF
+	if (VERBOSITY_LEVEL > 2)
+	{
+		domset_instance.printClusters();
+		domset_instance.exportToPLY(baseFileName + _T("_clusters.ply"));
+	}
+	#endif
+
+	const auto domset_clusters = domset_instance.getClusters();
+	VERBOSE("Clustering completed : %u clusters (%s)", domset_clusters.size(), TD_TIMER_GET_FMT().c_str());
+
+	for (int i = 0; i < domset_clusters.size(); ++i) 
+	{
+		const auto & cluster = domset_clusters[i];
+
+		Scene scene_cluster;
+		std::map<uint32_t, uint32_t> map_global_to_local;
+		std::vector<uint32_t> global_IDs;
+		scene_cluster.platforms = scene.platforms; // We copy all the plateforms for now, it's easier and harmless
+
+		uint32_t local_ID = 0;
+		for (const auto inCluster_ID : cluster)
+		{
+			const size_t global_ID = view_bkwd_reindexing[inCluster_ID];
+			auto & image = scene.images[global_ID];
+			image.ID = global_ID; 
+			scene_cluster.images.Insert(scene.images[global_ID]);
+			map_global_to_local[global_ID] = local_ID;
+			global_IDs.push_back(global_ID);
+			++local_ID;
+		}
+
+		FOREACH(IdxP, scene.pointcloud.points)
+		{
+			const auto & curr_viewArr = scene.pointcloud.pointViews[IdxP];
+			PointCloud::ViewArr new_viewArr;
+			
+			for(const auto & idxVG: global_IDs)
+			{
+				if(curr_viewArr.FindFirst(idxVG) != PointCloud::ViewArr::NO_INDEX)
+					new_viewArr.InsertSort(map_global_to_local.at(idxVG));
+			}
+
+			if(new_viewArr.GetSize() > 1)
+			{
+				scene_cluster.pointcloud.points.Insert(scene.pointcloud.points[IdxP]);
+				scene_cluster.pointcloud.pointViews.Insert(new_viewArr);
+				if(!scene.pointcloud.colors.IsEmpty())
+					scene_cluster.pointcloud.colors.Insert(scene.pointcloud.colors[IdxP]);
+				if(!scene.pointcloud.pointWeights.IsEmpty())
+					scene_cluster.pointcloud.pointWeights.Insert(scene.pointcloud.pointWeights[IdxP]);
+			}
+		}
+		LOG(String::FormatString("Saving cluster #%i", i));
+		scene_cluster.Save(baseFileName + String::FormatString("_cluster_%i.mvs", i), (ARCHIVE_TYPE)OPT::nArchiveType);
```

Comment:
here I choose convention over configuration. I do not give user the choice of the directory/suffix.

Reply:
it is fine